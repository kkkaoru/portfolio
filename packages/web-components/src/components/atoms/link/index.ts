export * from './link-included-image';
