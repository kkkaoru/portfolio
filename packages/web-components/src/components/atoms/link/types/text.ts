import { AnchorHTMLAttributes, DetailedHTMLProps } from 'react';

export type LinkTextProps = {
  text: string;
  href: string;
  target?: string;
  rel?: string;
} & DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>;
