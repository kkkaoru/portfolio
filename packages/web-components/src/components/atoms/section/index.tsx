type Props = {
  children: React.ReactNode;
};

export function Section({ children }: Props): JSX.Element {
  return <section className="max-w-full pt-6">{children}</section>;
}
// className = 'grid place-items-center';
