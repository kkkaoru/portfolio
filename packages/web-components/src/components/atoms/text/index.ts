export * from './text-with-underline';
