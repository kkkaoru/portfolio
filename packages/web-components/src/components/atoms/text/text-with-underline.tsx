export type Props = {
  text: string;
};

export function TextWithUnderline({ text }: Props): JSX.Element {
  return (
    <div className="flex flex-col items-center justify-center w-full p-10 font-bold break-all text-7xl lg:text-9xl after:block after:w-full after:h-1 after:bg-red-100">
      {text}
    </div>
  );
}
