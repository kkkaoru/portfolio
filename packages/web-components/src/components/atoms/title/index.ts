export * from './types';
export * from './h2';
