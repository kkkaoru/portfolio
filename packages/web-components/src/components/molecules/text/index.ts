export * from './title-and-description';
