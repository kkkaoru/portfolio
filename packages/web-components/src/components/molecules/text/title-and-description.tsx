import { H2, TitleProps } from '@web-components/components/atoms/title';
import { Description, DescriptionProps } from '@web-components/components/atoms/description';

export type TitleAndDescriptionProps = TitleProps & DescriptionProps;

export function TitleAndDescription({ title, description, useSerif }: TitleAndDescriptionProps): JSX.Element {
  return (
    <>
      <H2 title={title} />
      <Description description={description} useSerif={useSerif} />
    </>
  );
}
