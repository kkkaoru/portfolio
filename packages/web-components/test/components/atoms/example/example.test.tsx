import { render, screen } from '@testing-library/react';
import { TextWithUnderline } from '@web-components/components/atoms/text';

describe('example', () => {
  it('should be matched snapshot', () => {
    const { asFragment } = render(<TextWithUnderline text="snapshot" />, {});
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be included props text', () => {
    render(<TextWithUnderline text="will be included" />);
    expect(screen.getByText('will be included')).toBeInTheDocument();
  });
});
