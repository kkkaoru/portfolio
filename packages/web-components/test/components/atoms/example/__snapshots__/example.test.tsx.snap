// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example should be matched snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center w-full p-10 font-bold break-all text-7xl lg:text-9xl after:block after:w-full after:h-1 after:bg-red-100"
  >
    snapshot
  </div>
</DocumentFragment>
`;
